File: Problem 2 - Even Fibonacci Numbers/Python 3/problem_2.py
sequence = [0, 1]

# find all fibonacci sequence below a number
def fibonacci_sequence(below: int):
    next_number = 0
    while next_number < below:
        next_number = sequence[-1] + sequence[-2]
        if next_number < below:
            sequence.append(next_number)

# Sum all the even numbers in the sequence
def sum_even_numbers():
    sum = 0
    for number in sequence:
        if number % 2 == 0:
            sum += number
    return sum

if __name__ == '__main__':
    fibonacci_sequence(4000000)
    print(sum_even_numbers())

File: Problem 2 - Even Fibonacci Numbers/Java/Problem2.java
import java.util.ArrayList;
import java.util.List;

class Problem2 {

  static List<Integer> sequence = new ArrayList<Integer>();

  static void fibonacciSequence(int below) {
    sequence.add(0);
    sequence.add(1);
    int nextNumber = 0;
    while (nextNumber < below) {
      nextNumber = sequence.get(sequence.size() -1) + sequence.get(sequence.size() - 2);
      if (nextNumber < below) {
        sequence.add(nextNumber);
      }
    }
  }

  static int sumEvenFibonacciNumbers() {
    int sum = 0;
    for (int number : sequence) {
      if (number % 2 == 0) {
        sum += number;
      }
    }
    return sum;
  }

  public static void main(String[] args) {
    fibonacciSequence(4000000);
    System.out.println(sumEvenFibonacciNumbers());
  }

}
